pkg/ccl/backupccl/manifest_handling.go

```diff
@@ -193,12 +202,36 @@ func readBackupManifest(
 
 	fileType := http.DetectContentType(descBytes)
 	if fileType == ZipType {
-		descBytes, err = DecompressData(descBytes)
+		descBytes, err = decompressData(descBytes)
 		if err != nil {
 			return BackupManifest{}, errors.Wrap(
 				err, "decompressing backup manifest")
 		}
 	}
+
+	checksumFile, err := exportStore.ReadFile(ctx, filename+BackupManifestChecksumSuffix)
+	if err == nil {
+		// If there is a checksum file present, check that it matches.
+		defer checksumFile.Close()
+		checksumFileData, err := ioutil.ReadAll(checksumFile)
+		if err != nil {
+			return BackupManifest{}, errors.Wrap(err, "reading checksum file")
+		}
+		checksum, err := getChecksum(descBytes, encryptionKey)
```

Comment:
simply mixing the key in to the hash would cause a checksum mismatch if we use the wrong key which could be misleading. If we wanted to checksum the raw bytes, I'd probably actually just encrypt them like the other files.  (but i think i like #53298 better anyway)

Reply:
Ah, good point.